src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1095,15 +1109,26 @@ private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref
         {
             bool refersToNoPiaLocalType;
 
+            var isConstFound = false;
             SignatureTypeCode typeCode;
-            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
+            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode, acceptRequiredModifier: type =>
+            {
+                var isConst = IsConstModifierType(type);
+                isConstFound |= isConst;
+                return isConst;
+            });
```

Comment:
This lambda is capturing hence is adding two allocations to each call to `DecodeModifiersOrThrow`.  Is this a sensitive enough area of the code that we should consider a non-allocating solution?